src/parser_json.c

```diff
@@ -452,8 +452,18 @@ json_get_value(struct lyd_node_leaf_list *leaf, struct lyd_node **first_sibling,
         if (data[len++] != '[') {
             LOGVAL(ctx, LYE_XML_INVAL, LY_VLOG_LYD, leaf, "JSON data (expected begin-array)");
             return 0;
-        }
-
+        } 
+        unsigned int empty_list_len = len;
+        // Skip whitespaces
+        empty_list_len += skip_ws(&data[empty_list_len]);
+        // Check if an empty list [] was provided
+        if (data[empty_list_len] == ']') {
+            leaf->value_str = lydict_insert(ctx, "", 0);
```

Comment:
Setting "empty" value will not be enough, you want to free the whole `leaf` but that is probably not expected in this function so it will need more changes.